add: vue-shopify-account example (#92)
getnacelle/nacelle-js: examples/vue-shopify-account/README.md

If you want to have a better understanding of the queries and mutations that we are utlizing, check out the [Shopify API documentation](https://shopify.dev/api).

## Prerequisites

- Must be on the [Shopify Plus](https://www.shopify.com/plus/enterprise-ecommerce) plan and have [Multipass](https://help.shopify.com/en/api/reference/plus/multipass) enabled.
- You'll need your store's:
  - Storefront API token
  - Multipass secret

## Setup

1. add a few items to our `.env` file:

```sh
VITE_MYSHOPIFY_DOMAIN=""
VITE_SHOPIFY_TOKEN=""
VITE_SHOPIFY_ADMIN_PASSWORD=""
VITE_SHOPIFY_MULTIPASS_SECRET=""
```

this specific project uses vite for frontend tooling, in your own project if you aren't using vite, the `VITE_` prefix is unnecessary.

2. `npm install`

3. you will also need the [vercel CLI](https://vercel.com/cli) installed in order to test serverless functions

## Serverless Functions

To use Multipass and the address form for account pages this project relies on two packages:

- [multipassify](https://github.com/beaucoo/multipassify)
- [countrycitystatejson](https://github.com/khkwan0/countryCityStateJson)

> Both of these packages are _large_ and can add a lot to your client bundle, so using them only in serverless functions keeps the client a little more lightweight. But if you want another reason, it also keeps your Multipass secret outside of client code.

Reviewer: Nice call out about keeping the Multipass secret outside the client!

> This project includes a folders for serverless functions:

Reviewer: ```suggestion
This project includes a folder for serverless functions:
```

- Vercel in [api/\*](./api)

When testing your serverless functions locally make sure to use the platform's CLI ([Vercel CLI](https://vercel.com/docs/cli). NPM scripts for running the project with these CLI's are provided in the `package.json`.

## Functions

All the functions to interact with the API live in the `/components/AccountProvider.vue` file and the graphql queries are in the `/gql/index.js`.

> We are using [the composition API](https://vuejs.org/guide/extras/composition-api-faq.html) for the state management but this can be integrated into any state management system.

Reviewer: ```suggestion
We are using [the composition API](https://vuejs.org/guide/extras/composition-api-faq.html) for state management but this can be integrated into any state management system.
```

### Multipass Login

In the `/component/AccountProvider.vue` file there is a `multipassLogin` method, this will log in the user in Shopify. The best use for this function is to login the user before checkout, carrying over their information to the checkout page.

## Using Nuxt?

> If you implementing this system into a nuxt project we recommend making use of [middleware](https://nuxtjs.org/docs/directory-structure/middleware/) to protect authenticated routes:

Reviewer: ```suggestion
If you are implementing this system into a nuxt project we recommend using [middleware](https://nuxtjs.org/docs/directory-structure/middleware/) to protect authenticated routes:
```

Here is an example of what it might look like:

`middleware/authenticated.js`

```js
export default function (ctx) {
  const { store, redirect, app } = ctx
  // If the user is not authenticated
  const customerAccessToken = app.$cookies.get('customerAccessToken')
  if (customerAccessToken === undefined || customerAccessToken === false) {
    return redirect('/account/login')
  }
}
```

`middleware/notAuthenticated.js`

```js
export default function ({ store, redirect, app }) {
  // If the user is authenticated redirect to account
  const customerAccessToken = app.$cookies.get('customerAccessToken')

  if (customerAccessToken) {
    return redirect('/account')
  }
}
```

> These example assume the adition of the `cookie-universal-nuxt` [package](https://www.npmjs.com/package/cookie-universal-nuxt).

Reviewer: ```suggestion
These examples assume the addition of `cookie-universal-nuxt` [package](https://www.npmjs.com/package/cookie-universal-nuxt).
```
